File: EventDrivenStreamingPlatform/infrastructure/src/main/java/com/netflix/streaming/infrastructure/config/OpenApiConfig.java
package com.netflix.streaming.infrastructure.config;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import io.swagger.v3.oas.models.servers.Server;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.List;

/**
 * OpenAPI/Swagger configuration for API documentation.
 */
@Configuration
public class OpenApiConfig {

    @Bean
    public OpenAPI streamingPlatformOpenAPI() {
        return new OpenAPI()
                .info(new Info()
                        .title("Event-Driven Streaming Platform API")
                        .description("""
                            Complete Netflix-grade Event-Driven Architecture API for video streaming analytics.
                            
                            Features:
                            - Event-driven architecture with CQRS and Event Sourcing
                            - Real-time analytics and WebSocket streaming
                            - Machine learning pipeline orchestration
                            - Comprehensive compliance (GDPR, SOX)
                            - Production-grade observability
                            """)
                        .version("1.0.0")
                        .contact(new Contact()
                                .name("Streaming Platform Team")
                                .email("platform-team@example.com"))
                        .license(new License()
                                .name("Apache 2.0")
                                .url("https://www.apache.org/licenses/LICENSE-2.0")))
                .servers(List.of(
                        new Server()
                                .url("https://api.streaming-platform.com")
                                .description("Production Server"),
                        new Server()
                                .url("https://staging-api.streaming-platform.com")
                                .description("Staging Server"),
                        new Server()
                                .url("http://localhost:8080")
                                .description("Local Development Server")
                ));
    }
}

File: KotlinPaymentsPlatform/shared/src/main/kotlin/com/example/kotlinpay/shared/config/OpenApiConfig.kt
package com.example.kotlinpay.shared.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import io.swagger.v3.oas.models.servers.Server
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * OpenAPI/Swagger configuration for API documentation.
 */
@Configuration
class OpenApiConfig {

    @Bean
    fun paymentsPlatformOpenAPI(): OpenAPI {
        return OpenAPI()
            .info(
                Info()
                    .title("Kotlin Payments Platform API")
                    .description("""
                        Production-grade payment processing platform with comprehensive risk assessment,
                        secure payment processing, and enterprise monitoring.
                        
                        Features:
                        - Payment processing with multiple gateway support
                        - Real-time fraud detection and risk assessment
                        - Financial ledger and accounting
                        - PCI DSS Level 1 compliance
                        - Comprehensive audit logging
                    """.trimIndent())
                    .version("1.0.0")
                    .contact(
                        Contact()
                            .name("Payments Platform Team")
                            .email("payments-team@example.com")
                    )
                    .license(
                        License()
                            .name("Proprietary")
                            .url("https://example.com/license")
                    )
            )
            .servers(
                listOf(
                    Server()
                        .url("https://api.payments-platform.com")
                        .description("Production Server"),
                    Server()
                        .url("https://staging-api.payments-platform.com")
                        .description("Staging Server"),
                    Server()
                        .url("http://localhost:8080")
                        .description("Local Development Server")
                )
            )
    }
}

